x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/security/user/SecurityProfileUser.java

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.core.security.user;
+
+import org.elasticsearch.xpack.core.security.authz.RoleDescriptor;
+import org.elasticsearch.xpack.core.security.support.MetadataUtils;
+
+import java.util.Map;
+
+/**
+ * internal user that manages xpack security. Has all cluster/indices permissions.
+ */
+public class SecurityProfileUser extends User {
+
+    public static final String NAME = UsernamesField.SECURITY_PROFILE_NAME;
+    public static final SecurityProfileUser INSTANCE = new SecurityProfileUser();
+    private static final String ROLE_NAME = UsernamesField.SECURITY_PROFILE_ROLE;
+    public static final RoleDescriptor ROLE_DESCRIPTOR = new RoleDescriptor(
+        ROLE_NAME,
+        null,
+        new RoleDescriptor.IndicesPrivileges[] {
+            RoleDescriptor.IndicesPrivileges.builder()
+                .indices(".security-profile*")
```

Comment:
I think we should use the same index pattern from the `SecuritySystemIndices#getSecurityProfileIndexDescriptor()` .

Reply:
I didn't want to use regex because I was paranoid about performance. But it's probably trivial (since XPackUser has them). So I updated to match the pattern used by the system index.